### PR TITLE
EventLogger: add first source URL for events

### PR DESCRIPTION
**Pull request by attfarhan**
This was requested by BizOps to be able to see where anonymous users are coming from (even before they convert). Currently, we would only be able to see the Original Source for users that actually convert. With this, we can parse the URLs and see if we have anonymous users conducting cloud actions that come from ads, referrals etc.

@ebrodymoore tagging you as reviewer since unsure if you're still interested in this, but will tag devs to review the code if we want to move forward.

### CODE REVIEW

File: client/web/src/user/settings/backend.tsx [hunk 1]
```diff
@@ -145,6 +145,7 @@ export function logEvent(event: string, eventProperties?: unknown): void {
         {
             event,
             userCookieID: eventLogger.getAnonymousUserID(),
+            firstSourceURL: eventLogger.getFirstSourceURL(),
```
Comment (ebrodymoore) [inline client/web/src/user/settings/backend.tsx:148]: We'd need to add a column to the event_logs table for this, right?

**Review by ebrodymoore — APPROVED** [review]
Yes, absolutely still want this. I'm trusting you with the code and one clarifying question for how this is being executed. If my interpretation is correct, this is perfect!

> "This was requested by BizOps"

You are BizOps 😆

File: client/web/src/user/settings/backend.tsx (hunk 1 above)
Comment (attfarhan) [reply client/web/src/user/settings/backend.tsx:148]: Yeah

**Comment by attfarhan** [comment]
In the future, when we have private code, there could be a scenario where a non-signed up user gets linked to a private repo on Sourcegraph.com. If this is their first time landing on the page, this would mean this URL would get populated as their first source URL, meaning it may leak private repo info. This actually might mean we have to remove any first source URL logic from sourcegraph.com, and only populate it for info.*, about.*, etc.

**Comment by attfarhan** [comment]
But that might get more complicated with the move from about.sourcegraph.com to consolidating everything on sourcegraph.com

File: client/web/src/user/settings/backend.tsx (hunk 1 above)
Comment (ebrodymoore) [reply client/web/src/user/settings/backend.tsx:148]: Done

**Review by eseliger — APPROVED** [review]

File: cmd/frontend/graphqlbackend/schema.graphql [hunk 2]
```diff
@@ -418,6 +418,10 @@ type Mutation {
         """
         userCookieID: String!
         """
+        The first sourcegraph URL visited by the user, stored in a browser cookie.
+        """
+        firstSourceURL: String!
```
Comment (eseliger) [inline cmd/frontend/graphqlbackend/schema.graphql:423]: Should this be optional for backwards compatibility or doesn't this matter here?

**Review by eseliger — APPROVED** [review]

**Pull request merged**